feedback.py

```diff
@@ -0,0 +1,52 @@
+import asyncio
+from typing import Callable
+import discord
+from quest import queue, step
+
+# TODO -> Once Quest provides it use that instead
+
+class FeedbackWorkflow:
+    def __init__(self, record_feedback, post_event_function, send_message):
+        self._record_feedback = step(record_feedback)
+        self.post_event = post_event_function
+        self._send_message = step(send_message)
+
+    async def request_feedback(self, guild_id: int, thread_id: int, user_id: int):
+        async with queue("feedback", str(thread_id)) as feedback_queue:
+
+            feedback_view = FeedbackView(self.post_event, self._send_message, thread_id)
+            message_content = f"<@{user_id}>, on a scale of 1 to 5, how helpful was this conversation?"
+
+            await self._send_message(channel_id=thread_id, message=message_content, view=feedback_view)
+            try:
+                feedback_score = await asyncio.wait_for(feedback_queue.get(), timeout=300)
+                await self._send_message(thread_id, f'Thank you for your feedback!')
+                await self._record_feedback(guild_id, thread_id, user_id, feedback_score)
+            except asyncio.TimeoutError:
+                await self._send_message(thread_id, 'Feedback time out.')
+
+class FeedbackButton(discord.ui.Button):
+    def __init__(self, label, thread_id, post_event, _send_message):
+        super().__init__(label=label, style=discord.ButtonStyle.grey)
+        self.post_event = post_event
+        self._send_message = _send_message
+        self.thread_id = thread_id
+
+    async def callback(self, interaction: discord.Interaction): # Needs interaction for the function callback to work even if it is not used
+        feedback_score = self.label
+
+        await self.post_event(str(self.thread_id), "feedback", str(self.thread_id), "put", feedback_score)
```

Comment:
Instead of passing `post_event` all they way to `FeedbackButton`, create a new function in `FeedbackWorkflow` just after the queue is created and pass that in. Like we discussed in class.